pypeit/core/telluric.py

```diff
@@ -586,7 +588,25 @@ def unpack_orders(sobjs, ret_flam=False):
     return wave, flam, flam_ivar, flam_mask
 
 
+# TODO: This function needs to be revisited.  Better yet, it would useful to
+# brainstorm about whether or not it's worth revisiting the spec1d datamodel.
```

Comment:
Do you mean the `OneSpec` data model or the data in a `spec1d` file?
I presume the former, but just checking.

Reply:
And if it is the former, I'm open to modifying it for sure.

Reply:
I have thoughts on the `spec1d` datamodel, but realize that ship has sailed and there's likely no use in me yelling at it as it drifts out to sea.  Beyond that, this `general_spec_reader` basically always throws an error or a warning or some combination of both, which is disconcerting.

Reply:
Yes I agree general_spec_reader stinks, but the problem is that this routine needs to be able to read in either a spec1d file, or a coadd file, which is why it was coded in this way. When this code was written I don't think the Oned coadd code was in place yet.